boards/pico_explorer_base/README.md

```diff
@@ -11,31 +11,120 @@ board developed by the Raspberry Pi Foundation based on the RP2040 chip.
 
 First, follow the [Tock Getting Started guide](../../doc/Getting_Started.md)
 
+## Installing elf2uf2
+
+The Raspberry Pi Pico RP2040 uses UF2 files for flashing. Tock compiles to an ELF file.
+The `elf2uf2` utility is needed to transform the Tock ELF file into an UF2 file.
+
+To install `elf2uf2`, run the commands:
+
+```bash
+$ git clone https://github.com/raspberrypi/pico-sdk
+$ cd pico-sdk
+$ cd tools/elf2uf2
+$ mkdir build
+$ cd build
+$ cmake ..
+$ make
+$ sudo cp elf2uf2 /usr/local/bin
```

Comment:
Have you tried using https://github.com/JoNil/elf2uf2-rs ? Being able to install using cargo would allow for less platform-dependent installation instructions.

Reply:
Yes I did and I wanted to use that, but the problem is it has a lot of dependencies, including udev libraries. It seems that it also tries to mount and copy the file to the Pi. I am not sure what to do here.

```
error: failed to run custom build command for `libudev-sys v0.1.4`

Caused by:
  process didn't exit successfully: `/tmp/cargo-installsGVtyF/release/build/libudev-sys-1b2a95ff39822fea/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=LIBUDEV_NO_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG
  cargo:rerun-if-env-changed=PKG_CONFIG
  cargo:rerun-if-env-changed=LIBUDEV_STATIC
  cargo:rerun-if-env-changed=LIBUDEV_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_STATIC
  cargo:rerun-if-env-changed=PKG_CONFIG_ALL_DYNAMIC
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_PATH
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_LIBDIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=HOST_PKG_CONFIG_SYSROOT_DIR
  cargo:rerun-if-env-changed=PKG_CONFIG_SYSROOT_DIR

  --- stderr
  thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: "Could not run `\"pkg-config\" \"--libs\" \"--cflags\" \"libudev\"`\nThe pkg-config command could not be found.\n\nMost likely, you need to install a pkg-config package for your OS.\nTry `apt install pkg-config`, or `yum install pkg-config`,\nor `pkg install pkg-config`, or `apk add pkgconfig` depending on your distribution.\n\nIf you've already installed it, ensure the pkg-config command is one of the\ndirectories in the PATH environment variable.\n\nIf you did not expect this build to link to a pre-installed system library,\nthen check documentation of the libudev-sys crate for an option to\nbuild the library from source, or disable features or dependencies\nthat require pkg-config."', /home/codespace/.cargo/registry/src/github.com-1ecc6299db9ec823/libudev-sys-0.1.4/build.rs:38:41
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

It seems to work on mac out of the box and with and `apt-get` in Linux.

Reply:
My understanding from the brief read was the elf2uf2-rs only tries to copy files to the pi if you pass specific cli options, but I might have misunderstood

Reply:
Yes, but it does not have a feature to disable some dependencies. I added it to the makefile.